nemo/collections/asr/parts/utils/timestamp_utils.py

```diff
@@ -78,6 +78,7 @@ def segments_offset_to_time(segments, window_stride, subsampling_factor):
 
     for idx, hyp in enumerate(outputs):
         timestamp, text = extract_words_with_timestamps(hyp.text, subsampling_factor, window_stride)
+        text = text.strip()     # remove leading and trailing whitespaces - training data artifact
```

Comment:
instead of here, do this at the beginning of the `extract_words_with_timestamps` function.